src/pages/Earn/index.tsx

```diff
@@ -47,31 +60,88 @@ export default function Earn({
   const { chainId } = useActiveWeb3React()
   const { t } = useTranslation()
   const stakingInfos = useStakingInfo(Number(version))
+
   const [poolCards, setPoolCards] = useState<any[]>()
   const [filteredPoolCards, setFilteredPoolCards] = useState<any[]>()
   const [searchQuery, setSearchQuery] = useState<string>('')
+  const [sortBy, setSortBy] = useState<any>({ field: '', desc: true })
   const debouncedSearchQuery = useDebounce(searchQuery, 250)
+  const [stakingInfoData, setStakingInfoData] = useState<any[]>(stakingInfos)
 
   const stakingInfoV0 = useStakingInfo(Number(0))
-  const hasPositionV0 = stakingInfoV0?.some((stakingInfo) => stakingInfo.stakedAmount.greaterThan('0'))
+  const hasPositionV0 = stakingInfoV0?.some(stakingInfo => stakingInfo.stakedAmount.greaterThan('0'))
 
-  const handleSearch = useCallback((event) => {
+  const handleSearch = useCallback(event => {
     setSearchQuery(event.target.value.trim().toUpperCase())
   }, [])
 
   useEffect(() => {
-    const filtered = poolCards?.filter(card =>
-      card.props.stakingInfo.tokens[0].symbol.toUpperCase().includes(debouncedSearchQuery)
-      || card.props.stakingInfo.tokens[1].symbol.toUpperCase().includes(debouncedSearchQuery))
+    const filtered = poolCards?.filter(
+      card =>
+        card.props.stakingInfo.tokens[0].symbol.toUpperCase().includes(debouncedSearchQuery) ||
+        card.props.stakingInfo.tokens[1].symbol.toUpperCase().includes(debouncedSearchQuery)
+    )
     setFilteredPoolCards(filtered)
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [poolCards, debouncedSearchQuery])
 
+  useEffect(() => {
+    Promise.all(
+      stakingInfoData
+        ?.filter(function(info) {
+          // Only include pools that are live or require a migration
+          return !info.isPeriodFinished || info.stakedAmount.greaterThan(JSBI.BigInt(0))
+        })
+        .sort(function(info_a, info_b) {
+          if (sortBy.field === 'totalStakedInWavax') {
+            if (sortBy.desc) {
+              return info_a.totalStakedInWavax?.greaterThan(info_b.totalStakedInWavax ?? JSBI.BigInt(0)) ? -1 : 1
+            } else {
+              return info_a.totalStakedInWavax?.lessThan(info_b.totalStakedInWavax ?? JSBI.BigInt(0)) ? -1 : 1
+            }
+          }
+          if (sortBy.field === 'multiplier') {
```

Comment:
would these values be better as constant variables (enums)?

Reply:
fixed.